modifiers/caliper/modifier.py

```diff
@@ -42,7 +53,7 @@ class Caliper(SpackModifier):
 
     env_var_modification(
         "CALI_CONFIG",
-        "spot(output={}{})".format(_cali_datafile, "${CALI_CONFIG_MODE}"),
```

Comment:
@dyokelson We use `${CALI_CONFIG_MODE}` to enable (multiple) data collection modes like mpi, topdown, cuda etc. Will spot generate the correct output without this variable removed?

Reply:
Oh that was part of your new change, I had made these edits before merging that, I will add it back to have both. 

Also just FYI this PR is still a work in progress, there are still many changes needed.